### ISSUE TITLE
Erroneous "undefined variable" after specific method call

### ISSUE THREAD

**Issue by mrabarnett** [microsoft/pyright#11023]
Pylance version: 2025.8.3

OS: Linux Mint 21.3 Cinnamon 6.0.4

Visual Studio Code version: 1.105.0

Python version: 3.13

I'm using `pysmb` v1.2.13 and pylance is reporting an undefined variable.

The minimal (though unrunnable) code that shows this is:

```python
from smb.SMBConnection import SMBConnection

with SMBConnection(...) as con:
    con.connect(...)
    con.listPath(...) # <-- pylance reports "reportUndefinedVariable" for "con"
```

It's happy with the previous line, so...?

Incidentally, it's happy with this:
    
```python
from smb.SMBConnection import SMBConnection

with SMBConnection(...) as con:
    con.disconnect(...)
    con.listPath(...) # <-- No warning here.
```

and also with this:
    
```python
from smb.SMBConnection import SMBConnection

with SMBConnection(...) as con:
    con.foo()
    con.connect(...)
    con.listPath(...) # <-- No warning here.
```

**Comment by rchiodo** [comment]
I would hazard a guess that connect is coming out as always returning 'Never'. As in it throws an exception no matter what.

Oh yeah it seems that way:

<img width="1143" height="400" alt="Image" src="https://github.com/user-attachments/assets/b1c66891-a666-4e6a-b5b4-d7cfdfb5a8c3" />

So `con` is undefined after calling `connect`.

Seems like a bug though:

<img width="802" height="296" alt="Image" src="https://github.com/user-attachments/assets/c0973dd5-0ae0-48d4-a448-55dc8131d042" />

Not sure how auth_result is 'Never' here. 

Transferring to Pyright.

**Comment by rchiodo** [comment]
To reproduce, `pip install pysmb` and use the code provided by the original author.
